MinPlatformPkg/Test/Library/TestPointCheckLib/StandaloneMmTestPointCheckLib.c

```diff
@@ -0,0 +1,482 @@
+
+  Result = TRUE;
+  Status = TestPointCheckSmrr ();
+  if (EFI_ERROR (Status)) {
+    Result = FALSE;
+  }
+
+  if (Result) {
+    TestPointLibSetFeaturesVerified (
+      PLATFORM_TEST_POINT_ROLE_PLATFORM_IBV,
+      NULL,
+      5,
+      TEST_POINT_BYTE6_SMM_END_OF_DXE_SMRR_FUNCTIONAL
+      );
+  }
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmEndOfDxe - Exit\n"));
+  return EFI_SUCCESS;
+}
+
+/**
+  This service verifies the security of Standalone MM communication buffers at Standalone MM Ready To Lock.
+
+  Test subject: Standalone MM communication buffer.
+  Test overview: Verify only CommBuffer and MMIO are mapped in the page table.
+  Reporting mechanism: Dumps the memory map and GCD map at StandaloneMmReadyToLock and checks at StandaloneMmReadyToBoot.
+
+  @retval EFI_SUCCESS         The test point check was performed successfully.
+  @retval EFI_UNSUPPORTED     The test point check is not supported on this platform.
+**/
+EFI_STATUS
+EFIAPI
+TestPointStandaloneMmReadyToLockSecureStandaloneMmCommunicationBuffer (
+  VOID
+  )
+{
+  if ((mFeatureImplemented[TEST_POINT_INDEX_BYTE6_SMM] & TEST_POINT_BYTE6_SMM_READY_TO_LOCK_SECURE_SMM_COMMUNICATION_BUFFER) == 0) {
+    return EFI_SUCCESS;
+  }
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmReadyToLockSecureStandaloneMmCommunicationBuffer - Enter\n"));
+
+  //
+  // Collect information here, because it is last chance to access outside SMRAM.
+  //
+  // Previous memory collection is allowed in StandaloneMm but leaving this function for possible future collection
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmReadyToLockSecureStandaloneMmCommunicationBuffer - Exit\n"));
+  return EFI_SUCCESS;
+}
+
+/**
+  This service verifies the validity of the Standalone MM page table at Standalone MM Ready To Boot.
+
+  Test subject: Standalone MM page table.
+  Test overview: Verify the Standalone MM page table matches the Standalone MM memory attribute table.
+  Reporting mechanism: Set ADAPTER_INFO_PLATFORM_TEST_POINT_STRUCT.
+                       Reports an error message upon checking.
+
+  @retval EFI_SUCCESS         The test point check was performed successfully.
+  @retval EFI_UNSUPPORTED     The test point check is not supported on this platform.
+**/
+EFI_STATUS
+EFIAPI
+TestPointStandaloneMmReadyToBootStandaloneMmPageProtection (
+  VOID
+  )
+{
+  EFI_STATUS  Status;
+  BOOLEAN     Result;
+
+  if ((mFeatureImplemented[TEST_POINT_INDEX_BYTE6_SMM] & TEST_POINT_BYTE6_SMM_READY_TO_BOOT_SMM_PAGE_LEVEL_PROTECTION) == 0) {
+    return EFI_SUCCESS;
+  }
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmReadyToBootStandaloneMmPageProtection - Enter\n"));
+
+  Result = TRUE;
+
+  Status = TestPointCheckStandaloneMmPaging ();
+  if (EFI_ERROR (Status)) {
+    Result = FALSE;
+  }
+
+  if (Result) {
+    TestPointLibSetFeaturesVerified (
+      PLATFORM_TEST_POINT_ROLE_PLATFORM_IBV,
+      NULL,
+      6,
```

Comment:
And this? The smm instance has `TEST_POINT_INDEX_BYTE6_SMM` for all fields.

Reply:
Fixed.